csrc/alias_analysis.cpp

```diff
@@ -248,6 +248,11 @@ void AliasFinder::handle(const LoadStoreOp* set) {
   }
   TensorView* out = set->out()->as<TensorView>();
 
+  // The set operation is an inplace update and shouldn't be marked as alias op.
```

Comment:
For now I think it's only applicable to `set`.

Reply:
Why not `view`? For example, 

```
out = reshape(in, [6], [2, 3]);
out = add(out, 1)
out = reshape(out, [2, 3], [6]);
aliasOutputToInput(out, in);
```

Reply:
That's a great example.

It's an implementation thing I think. Yes. in theory we **could** support it. Since it's just a view and we should be able to infer the new stride for `out` when it's an alias of `in`.

But I don't think we do that today. https://github.com/NVIDIA/Fuser/blob/7b9271640b3b918b5cb65d9c68ea2c17d14319d2/csrc/runtime/allocations.cpp#L260-L275
We are blindly using `in` at the location of `out` in the allocation, which requires `out` and `in` to have the same size/stride.

I'll try your example and make sure that an error is thrown for that.

Reply:
> which requires out and in to have the same size/stride.

`in` and `out` in my example do have the same size/stride. There's a *roundtrip* of reshapes.

Reply:
oh how blind was I :face_with_head_bandage: 

So I think your earlier comment makes sense, if we do this in `AliasAnalysisResult::finalize` and not materialize `alias` when it has a re-use buffer. We are not restricted by the op type.

Thanks for pointing it out. :saluting_face: